src/FlatSharp.Compiler/FlatSharpCompiler.cs

```diff
@@ -263,25 +264,53 @@ internal static byte[] GetBfbs(CompilerOptions options)
             Directory.CreateDirectory(outputDir);
             FileInfo info = new FileInfo(options.InputFile);
 
-            System.Diagnostics.Process p = new System.Diagnostics.Process
+            using var p = new Process
             {
-                StartInfo = new System.Diagnostics.ProcessStartInfo
+                StartInfo = new ProcessStartInfo
                 {
                     CreateNoWindow = true,
                     RedirectStandardError = true,
                     RedirectStandardOutput = true,
                     UseShellExecute = false,
                     FileName = flatcPath,
-                    Arguments = $"-b --schema --bfbs-comments --bfbs-builtins --bfbs-filenames \"{info.DirectoryName}\" --no-warnings -o \"{outputDir}\" \"{info.FullName}\"",
                 }
             };
 
+            foreach (var arg in new[]
+            {
+                "-b",
+                "--schema",
+                "--bfbs-comments",
+                "--bfbs-builtins",
+                "--bfbs-filenames",
+                info.DirectoryName,
+                "--no-warnings",
+                "-o",
+                outputDir,
+                info.FullName
+            })
+            {
+                p.StartInfo.ArgumentList.Add(arg);
+            }
+
             try
             {
-                p.Start();
-                string stdout = p.StandardOutput.ReadToEnd();
-                string stderr = p.StandardError.ReadToEnd();
+                p.EnableRaisingEvents = true;
+
+                var lines = new List<string>();
+
+                void OnDataReceived(object sender, DataReceivedEventArgs args)
+                {
+                    if (args.Data is { } line)
```

Comment:
I'm not familiar with the `is { } line` syntax. I've always seen `is SomeType x`.

Reply:
The `is { }` part is basically a [property pattern](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/patterns#property-pattern) and because the pattern doesn't restrict the match to any properties of the object, it ends up matching _any_ non-null value/string. The `line` part introduces a pattern variable that's initialized with the value of the match. It's another way of saying `is string line` but repeating the type seems redundant since [`DataReceivedEventArgs.Data`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.datareceivedeventargs.data) is always a string and what we're interested in testing here is its value, not its type. The property pattern is generally more applicable and `{ }` gets more interesting when you want to be able to test other aspects of the string such as its length; e.g. `is { Length: > 0 } line` will match only non-empty strings.

Reply:
Thanks! I learned something :)

Since this is really just a fancy null check, I'd rather use `!string.IsNullOrEmpty(s)`, since that makes the intent of the check more obvious.

Reply:
Done with a8cf3e051cdceb456b1262b2f2fbddf8abe2fa8c.